presto-cli/src/main/java/com/facebook/presto/cli/Console.java

```diff
@@ -320,6 +320,13 @@ private static boolean process(QueryRunner queryRunner, String sql, OutputFormat
                 schemaChanged.run();
             }
 
+            //update path if present
+            if (query.getSetPath().isPresent()) {
```

Comment:
Would not be better to create `ClientSession.Builder` in line 312 and just update path here instead of creating new `session`, then in line 360 you could just build a session.